src/engine/SPARQLProtocol.h

```diff
@@ -205,5 +271,11 @@ class SPARQLProtocol {
     return tokenFromAuthorizationHeader
                ? std::move(tokenFromAuthorizationHeader)
                : std::move(tokenFromParameter);
-  };
+  }
+
+  // Extract the graph to be acted upon using from the URL query parameters
+  // (`Indirect Graph Identification`). See
+  // https://www.w3.org/TR/2013/REC-sparql11-http-rdf-update-20130321/#indirect-graph-identification
+  static GraphOrDefault extractTargetGraph(
+      const ad_utility::url_parser::ParamValueMap& params);
```

Comment:
That function is much too long (before it was borderline, now definitely)
I think the two cases that are splittable are `GET` requests and `POST` requests which both have their own logic with many details.
That also makes the testing easier.